app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div>
+    <v-form>
+      <v-row>
+        <v-col cols="12" xl="4">
+          <v-text-field
+            dense
+            flat
+            label="Name"
+            data-test="text-formName"
+            v-model="formName"
+          />
+        </v-col>
+        <v-col cols="12" xl="8">
+          <v-text-field
+            dense
+            flat
+            label="Description"
+            data-test="text-formDescription"
+            v-model="formDescription"
+          />
+        </v-col>
+      </v-row>
+      <v-btn color="primary" @click="submitFormSchema">
+        <span>Save Form Design</span>
+      </v-btn>
+    </v-form>
+    <FormBuilder :form="formSchema" @change="onChangeMethod" :options="{}" />
+  </div>
+</template>
+
+<script>
+import { FormBuilder } from 'vue-formio';
+import formService from '@/services/formService';
+
+export default {
+  name: 'FormDesigner',
+  components: {
+    FormBuilder
+  },
+  data: () => ({
+    formName: '',
+    formDescription: '',
+    formSchema: {}
+  }),
+  created() {
+    this.getFormSchema();
+  },
+  methods: {
+    async getFormSchema() {
+      this.formSchema = {};
+      try {
+        if (this.formId) {
+          const form = await formService.readForm(this.formId);
```

Comment:
you could just read the form, and look in its versions[] for the version you want.

Reply:
I did see that in the form level object response. The reason I didn't put in any custom logic to parse/list available versions is because I was assuming there would either be a management page or some other entrypoint to this component that would answer the question for it already. The current incarnation of this component is relatively idempotent - if it knows the `formId` and `formVersionId`. Not sure on how we handle the "create a totally new form" just yet - for later discussion.